general/development/policies/codingstyle/frankenstyle.md

```diff
@@ -0,0 +1,173 @@
+| Conditional availability | core_availability | /availability |
+| Backup and restore | core_backup | /backup/util/ui |
+| Badges | core_badges | /badges |
+| Blocks | core_block | /blocks |
+| Blogging | core_blog | /blog |
+| Bulk users operations | core_bulkusers |  |
+| Caching | core_cache | /cache |
+| Calendar | core_calendar | /calendar |
+| Cohorts | core_cohort | /cohort |
+| Comment | core_comment | /comment |
+| Competency based education | core_competency | /competency |
+| Completion | core_completion | /completion |
+| Countries | core_countries |  |
+| Course | core_course | /course |
+| Currencies | core_currencies |  |
+| Database transfer | core_dbtransfer |  |
+| Debugging | core_debug |  |
+| Text editors | core_editor | /lib/editor |
+| Education fields | core_edufields |  |
+| Enrol | core_enrol | /enrol |
+| Error reporting | core_error |  |
+| Favourites | core_favourites | /favourites |
+| File picker | core_filepicker |  |
+| Files management | core_files | /files |
+| User filtering | core_filters |  |
+| Forms | core_form | /lib/form |
+| Grades | core_grades | /grade |
+| Advanced grading | core_grading | /grade/grading |
+| Groups | core_group | /group |
+| Help | core_help |  |
+| Hub | core_hub |  |
+| IMS CC | core_imscc |  |
+| Installer | core_install |  |
+| ISO 6392 | core_iso6392 |  |
+| Language pack configuration | core_langconfig |  |
+| License | core_license |  |
+| Maths library | core_mathslib |  |
+| Media | core_media |  |
+| Messaging | core_message | /message |
+| MIME types | core_mimetypes |
+| MNet | core_mnet | /mnet |
+| Dashboard | core_my | /my |
+| User notes | core_notes | /notes |
+| Page types | core_pagetype |  |
+| Pictures and icons | core_pix |  |
+| Plagiarism | core_plagiarism | /plagiarism |
+| Plugins management | core_plugin |  |
+| Portfolio | core_portfolio | /portfolio |
+| Privacy | core_privacy | /privacy |
+| Course publishing | core_publish | /course/publish |
+| Question bank engine | core_question | /question |
+| Ratings | core_rating | /rating |
+| Site registration | core_register | /admin/registration |
+| Repository | core_repository | /repository |
+| RSS | core_rss | /rss |
+| Roles | core_role | /admin/roles |
+| Global search | core_search | /search |
+| Tabular data display/download (deprecated) | core_table |  |
+| Tagging | core_tag | /tag |
+| Timezones | core_timezones |  |
+| User | core_user | /user |
+| User key | core_userkey |  |
+| Web service | core_webservice | /webservice |
+
+## Usage
+
+Frankenstyle component names are used in:
+
+### Function names
+
+All plugin functions must start with full frankenstyle prefix. For backwards compatibility modules may also use **modulename_** as prefix.
+
+### Class names
+
+- All the component classes must sit under the classes ([auto-loaded](https://docs.moodle.org/dev/Automatic_class_loading)) directory, belonging to their frankenstyle namespace and having a natural name, ex: \mod_forum\example.
```

Comment:
Don't use `ex:`. This may not be easy to understand for some readers who are not native English speakers.

```suggestion
All component classes must be placed under the classes directory, which allows them to be ([autoloaded](https://docs.moodle.org/dev/Automatic_class_loading)). These should be placed in a namespace according to their frankenstyle component name, and having a natural class name, for example a discussion class in the forum activity should be in the `mod_forum` namespace and may have a class name of `dicussion` - `\mod_forum\discussion`.

```